### PR TITLE
RollupParams refactor + STR 478 functional test for full node

### PR DESCRIPTION
**Pull request by voidash**
## Description

* Resolves the comment present in https://github.com/alpenlabs/strata/pull/343#discussion_r1786911020
* Add functional test for Full node sync

### Type of Change

-   [ ] Bug fix (non-breaking change which fixes an issue)
-   [ ] New feature/Enhancement (non-breaking change which adds functionality or enhances an existing one)
-   [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
-   [ ] Documentation update
-   [ ] Refactor

## Checklist

-   [x] I have performed a self-review of my code.
-   [x] I have commented my code where necessary.
-   [x] I have updated the documentation if needed.
-   [x] My changes do not introduce new warnings.
-   [x] I have added tests that prove my changes are effective or that my feature works.
-   [x] New and existing tests pass with my changes.

### CODE REVIEW

File: functional-tests/constants.py [hunk 1]
```diff
@@ -47,8 +45,23 @@
         "timeout": 30,
     },
     "max_deposits_in_block": 16,
-    "operator_config": {"static": [{"signing_pk": "01" * 32, "wallet_pk": "02" * 32}]},
+    "operator_config": {
+        "static": [
+            {
+                "signing_pk": "0x5eaae68d0681f944a79567bbf96b5deda944d5287d898b744a100487dc28e5d5",
+                "wallet_pk": "0x5276d54e024127b290462c7eda211eb576332fc9a6b9a9479d7d152b5047193a",
```
[the remark this answers is not in the document]
Comment (voidash) [reply functional-tests/constants.py:52]: i just copied it from datatool output. So i think it doesn't matter

File: crates/primitives/src/params.rs [hunk 2]
```diff
@@ -31,6 +31,9 @@ pub struct RollupParams {
     /// Config for how the genesis operator table is set up.
     pub operator_config: OperatorConfig,
 
+    /// Config for various transaction
```
Comment (storopoli) [inline crates/primitives/src/params.rs:34]:
```suggestion
    /// Config for various transactions
```

File: crates/tx-parser/src/deposit/test_utils.rs [hunk 3]
```diff
@@ -22,12 +22,14 @@ pub fn test_taproot_addr() -> BitcoinAddress {
     BitcoinAddress::parse(&addr.to_string(), bitcoin::Network::Regtest).unwrap()
```
Comment (storopoli) [inline crates/tx-parser/src/deposit/test_utils.rs:22]:
```suggestion
    BitcoinAddress::parse(&addr.to_string(), Network::Regtest).unwrap()
```
can we move this import up.

File: crates/tx-parser/src/filter.rs [hunk 4]
```diff
@@ -21,16 +20,13 @@ pub enum TxFilterRule {
...
     /// Deposit transaction with deposit config and address
-    Deposit(DepositTxParams),
+    Deposit(DepositTxConfig),
 }
 
 type RollupName = String;
 
-/// Reads the operator wallet public keys from Rollup params. Returns None if
-/// not yet bootstrapped
-/// FIXME: This is only for devnet as these pks have to be read from the chain state
```
Comment (storopoli) [inline crates/tx-parser/src/filter.rs:33]: Why are we removing this docstrings?

File: crates/tx-parser/src/utils.rs [hunk 5]
```diff
@@ -1,12 +1,11 @@
...
 use bitcoin::{
     opcodes::all::OP_PUSHNUM_1,
     script::{Instruction, Instructions},
     taproot::TaprootBuilder,
-    Address, Network, Opcode, XOnlyPublicKey,
+    Address, Network, Opcode,
 };
 use musig2::{
-    secp256k1::{PublicKey, SECP256K1},
+    secp256k1::{PublicKey, XOnlyPublicKey, SECP256K1},
```
Comment (storopoli) [inline crates/tx-parser/src/utils.rs:8]: Good this makes sure that our pks x-only pks are always `musig2` compatible.

**Comment by delbonis** [comment]
@prajwolrg Would definitely want to cache it in the chainstate somehow.  We also would be able to reuse a bunch of state across an epoch, and that would be used across a range of L1 blocks since we will only do the L1 checkin at the end of the epoch.

File: bin/strata-client/src/rpc_client.rs [hunk 6]
```diff
@@ -1,8 +1,7 @@
...
+use jsonrpsee::http_client::{HttpClient, HttpClientBuilder};
 
-pub async fn sync_client(url: &str) -> Client {
-    WsClientBuilder::default()
+pub fn sync_client(url: &str) -> HttpClient {
+    HttpClientBuilder::default()
         .build(url)
-        .await
         .expect("Failed to connect to the RPC server")
 }
```
Comment (sapinb) [inline bin/strata-client/src/rpc_client.rs:7]: @voidash Can you make another PR with just changes related to this fix without the other param changes ?

Comment (voidash) [reply bin/strata-client/src/rpc_client.rs:7]: sure

File: functional-tests/fn_run_stuff.py
Comment (delbonis) [inline functional-tests/fn_run_stuff.py:1]: Stale file?

File: functional-tests/fn_full_node_sync.py
Comment (delbonis) [inline functional-tests/fn_full_node_sync.py:1]: This should be named something like `fn_sync_flaky_peers`.  The point of the test isn't that we're testing full node sync, we're trying to that we can *keep* syncing as connections are interrupted.  The first component of the test name is supposed to be some term describing what *function* it pertains to.  It's also good to add a comment at the start of the file that describes what the test does.

It might actually make sense to split it into 2 or 3 tests that test the different parties restarting.

File: functional-tests/fn_bridge_deposit.py [hunk 7]
```diff
@@ -95,6 +101,4 @@ def do(self, ctx: flexitest.RunContext, evm_addr: str):
 
         balance = int(rethrpc.eth_getBalance(f"0x{evm_addr}"), 16)
         net_balance = balance - original_balance
-        assert (
-            net_balance == deposit_amount
-        ), f"deposit processed multiple times, extra: {balance - original_balance - deposit_amount}"
+        assert balance == deposit_amt * SATS_TO_WEI, f"deposit processed multiple times: {balance}"
```
Comment (delbonis) [inline functional-tests/fn_bridge_deposit.py:104]: This kinda seems like it's duplicating the above checks?  Should specify that that was the amount accepted and what the expected amount was.

**Review by delbonis — CHANGES_REQUESTED** [review]
Should clean up the tests a bit and make them more granular so we have better coverage.

**Comment by voidash** [comment]
Closing this. As the changes relating to the full node sync has been fixed and the refactor part should rather make sense in main for which seperate PR will be created
